src/proxy.rs

```diff
@@ -34,14 +33,14 @@ impl ProxyType {
 ///
 /// If the given TokenStream is not valid, errors are emitted as appropriate
 /// and `Err(())` is returned.
-pub(crate) fn parse_types(args: TokenStream) -> Result<Vec<ProxyType>, ()> {
+pub(crate) fn parse_types(args: TokenStream) -> Vec<ProxyType> {
     let mut out = Vec::new();
     let mut iter = args.into_iter().peekable();
 
     // While there are still tokens left...
     while iter.peek().is_some() {
         // First, we expect one of the proxy types.
-        out.push(eat_type(&mut iter)?);
+        eat_type(&mut iter).map(|ty| out.push(ty));
```

Comment:
I don't really like to use `map` for non-pure function. An `if let` is better here IMO.